app/services/immediate_email_generation_service/batch.rb

```diff
@@ -27,19 +27,16 @@ def email_parameters
       @email_parameters ||= begin
         subscriptions_to_fulfill.map do |subscription|
           {
-            subscriber: subscription.subscriber,
             content: content_change || message,
-            subscriptions: [subscription],
+            subscription: subscription,
           }.compact
         end
       end
     end
 
     def subscription_content_records(email_ids)
       email_parameters.flat_map.with_index do |params, index|
-        params[:subscriptions].map do |subscription|
-          { subscription_id: subscription.id, email_id: email_ids[index] }
-        end
+        { subscription_id: params[:subscription].id, email_id: email_ids[index] }
```

Comment:
Hmm this looks like a subtle change. It looks like where previously we'd have had every subscription inserted as a subscription content this would just be the first one.

I guess this doesn't have any real world impacts. Could, only be a touch confusing if you were debugging but probably just another warning flag about subscription content's usage/purpose.

Reply:
Oh yes, I remember thinking that and realise I didn't write it anywhere. I've amended the third commit message to explain how it has no effect on the processing. I suppose it could lead to an interesting debugging problem if someone was querying _from_ a particular subscription, expecting to find an email that may actually have been associated with another. I agree it's mostly a signal of pain with the model itself, rather than the processing.

I've made a note on [the pain card about it](https://trello.com/c/0ge1rXr5/314-dev-pain-the-subscriptioncontent-model-is-confusing-and-hard-to-work-with).